runtime/nls/j9cl/j9jcl.nls

```diff
@@ -555,3 +555,35 @@ J9NLS_JCL_CRIU_MAX_RETRY_FOR_NOTCHECKPOINTSAFE_REACHED.explanation=CRIUSupport::
 J9NLS_JCL_CRIU_MAX_RETRY_FOR_NOTCHECKPOINTSAFE_REACHED.system_action=The JVM will throw a JVMCheckpointException.
 J9NLS_JCL_CRIU_MAX_RETRY_FOR_NOTCHECKPOINTSAFE_REACHED.user_response=View CRIU documentation to determine how to resolve the error.
 # END NON-TRANSLATABLE
+
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR_2=Could not close the checkpoint data directory, errno=%li
+# START NON-TRANSLATABLE
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.sample_input_1=1
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.explanation=An error occured when the JVM attempted to close the specified directory.
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.system_action=The JVM will throw a SystemRestoreException or SystemCheckpointException.
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.user_response=Ensure the specified directory is accessible.
+# END NON-TRANSLATABLE
+
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_WORK_DIR_2=Could not close the work directory, errno=%li
+# START NON-TRANSLATABLE
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_WORK_DIR.sample_input_1=1
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_WORK_DIR.explanation=An error occured when the JVM attempted to close the specified directory.
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_WORK_DIR.system_action=The JVM will throw a SystemRestoreException or SystemCheckpointException.
+J9NLS_JCL_CRIU_FAILED_TO_CLOSE_WORK_DIR.user_response=Ensure the specified directory is accessible.
+# END NON-TRANSLATABLE
+
+J9NLS_JCL_CRIU_FAILED_TO_RUN_INTERNAL_RESTORE_HOOKS_2=Could not run internal restore hooks, errno=%li
```

Comment:
Any reason not to use the existing `J9NLS_JCL_CRIU_FAILED_TO_RUN_INTERNAL_RESTORE_HOOKS` w/ the message content modification?

Reply:
Similar question for others like `J9NLS_JCL_CRIU_FAILED_DELAY_LOCK_RELATED_OPS`.

Reply:
Same reason we add tracepoints at the end of files. I believe the NLS tooling requires messages to be consistent or you need to modify all the translations

Reply:
NLS is different from the tracepoint, and there is no obsoleted process to support messages in earlier VM releases. I think the dev only add/modify the default (English) version, and the translations are taken care by another team (ID?), maybe that has been changed.
@pshipton any insights?

Reply:
See the guidance at the top of the file:
```
# New messages MUST be added at the end of this file.
# DO NOT delete messages from this file, as it will change their indices.
# If you wish to remove a message, delete its text, but leave the key in place
```

Reply:
If the messages haven't been translated yet, you could change the text. There is no active translation going on atm.

Reply:
That said, the proper open source behaviour is to follow the rules, rather than knowing the IBM translation schedule.

Reply:
Existing NLS:
```
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR=Could not close the checkpoint data directory, errno=%li
# START NON-TRANSLATABLE
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.sample_input_1=1
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.explanation=An error occurred when the JVM attempted to close the specified directory.
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.system_action=The JVM will throw a RestoreException or SystemCheckpointException.
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.user_response=Ensure the specified directory is accessible.
# END NON-TRANSLATABLE
```

New NLS:
```
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR_2=Could not close the checkpoint data directory, errno=%li
# START NON-TRANSLATABLE
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.sample_input_1=1
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.explanation=An error occured when the JVM attempted to close the specified directory.
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.system_action=The JVM will throw a SystemRestoreException or SystemCheckpointException.
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR.user_response=Ensure the specified directory is accessible.
# END NON-TRANSLATABLE
```

Can the existing NLS message in question be modified instead? If so, are the translation required to be updated at same time?

Reply:
J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR has been translated, so it shouldn't be modified in any meaningful way, in particular the parameters. You could correct spelling or grammar or other minor updates. The translated messages will remain the same until the next translation cycle, and then may change to reflect updates to the English version. 

The rules are still valid, going open source doesn't change them. IBM (or other open source consumers?) may pick up the explanation & system_action and publish them in a document. They are just there and are copied into the translations. Semeru doesn't publish them atm. For somebody to find them, they'd have to look in the source code for the message text, and read the following explanation & system_action. I suppose you'd expect the message in the head stream to match earlier releases, rather than looking at the exact source for the release you are using. If your message isn't in English, you'd be looking at the translated file and finding the explanation and system_action there. 

I expect we could get away with modifying the J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR system_action in place, in particular considering CRIU is only a preview so far. The nice thing to do is update the system_action in all the translations. It's still in English so each to do with search & replace.

Reply:
Thanks for the clarification, I guess the suggestion can be applied to `J9NLS_JCL_CRIU_FAILED_TO_RUN_INTERNAL_RESTORE_HOOKS` and `J9NLS_JCL_CRIU_FAILED_DELAY_LOCK_RELATED_OPS` in a similar fashion.